assets/js/search.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Javascript specially for the IaC Library page.
+ */
+
+(function () {
+
+  // Returns a function, that, as long as it continues to be invoked, will not
+  // be triggered. The function will be called after it stops being called for
+  // N milliseconds. If `immediate` is passed, trigger the function on the
+  // leading edge, instead of the trailing.
+  // From: https://davidwalsh.name/javascript-debounce-function
```

Comment:
I didn't understand anything about this comment until I went to the website, which contains a very good examplation: "ensures a given task doesn't fire so often that it bricks browser performance". I'd add this to the comment.
```suggestion
  // Ensures a given task doesn't fire so often that it bricks browser performance.
  // From: https://davidwalsh.name/javascript-debounce-function
```